app/controllers/users_controller.rb

```diff
@@ -0,0 +1,34 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show, :friends]
+  before_action :set_current_user, only: [:me, :edit]
+
+
+  def index
+    friends               = current_user.friends.ids
+    sent_request_to       = current_user.friend_requests_sent.pending.pluck(:receiver_id)
+    received_request_from = current_user.friend_requests_received.pending.pluck(:sender_id)
+    friend_or_has_pending_request = friends + [current_user.id] + sent_request_to + received_request_from
+    @users = User.all.where.not("id IN (?)", friend_or_has_pending_request)
```

Comment:
We would normally want our controllers not to be handling advanced db queries (outside of simple `find` and `find_by`). I suggest writing this logic in the `User` model instead (as either a scope or a class method) and then just invoke that here:

```ruby
@users = User.advanced_query_we_transferred_to_the_model
```